From f1ca3df5bad21b70c83435eda56fcabe6ec46760 Mon Sep 17 00:00:00 2001
From: Junxiao Chang <junxiao.chang@intel.com>
Date: Mon, 9 Aug 2021 15:48:12 +0800
Subject: [PATCH] drm/i915: disable guc submission

This is to fix CVE-2020-12364, only solution is to disable guc
submission.

Signed-off-by: Junxiao Chang <junxiao.chang@intel.com>
---
 drivers/gpu/drm/i915/intel_uc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_uc.c b/drivers/gpu/drm/i915/intel_uc.c
index 7c95697e1a35..103e18baa2cc 100644
--- a/drivers/gpu/drm/i915/intel_uc.c
+++ b/drivers/gpu/drm/i915/intel_uc.c
@@ -64,6 +64,9 @@ static int __get_platform_enable_guc(struct drm_i915_private *i915)
 
 	/* Any platform specific fine-tuning can be done here */
 
+	/* CVE-2020-12364 fix - disable guc submission */
+	enable_guc &= ~(ENABLE_GUC_SUBMISSION);
+
 	return enable_guc;
 }
 
-- 
2.17.1

